File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):

        first_word = strs[0]
        prefix = ""
        i = 0

        if len(strs) == 1: return first_word

        while i < len(first_word):
            for e in strs[1::]:
                if len(e) <= i or first_word[i] != e[i]:return prefix
            prefix += first_word[i]
            i += 1
        return prefix
            


        

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }

        string prefix;
        string firstWord = strs[0];
        for (int i = 0; i < firstWord.length(); i++) {
            for (string str : strs) {
                if (str[i] !=   firstWord[i]) {return prefix;}
            }
            prefix += firstWord[i];
        }

        return prefix;
    }
};